File: java/l2server/gameserver/network/serverpackets/ChangeMoveType.java
/*
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
 * details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see <http://www.gnu.org/licenses/>.
 */

package l2server.gameserver.network.serverpackets;

import l2server.gameserver.model.actor.L2Character;

/**
 * sample
 * <p>
 * 0000: 3e 2a 89 00 4c 01 00 00 00						 .|...
 * <p>
 * format   dd
 *
 * @version $Revision: 1.3.2.1.2.4 $ $Date: 2005/03/27 15:29:57 $
 */
public class ChangeMoveType extends L2GameServerPacket
{
	public static final int WALK = 0;
	public static final int RUN = 1;

	private int _charObjId;
	private boolean _running;

	public ChangeMoveType(L2Character character)
	{
		_charObjId = character.getObjectId();
		_running = character.isRunning();
	}

	@Override
	protected final void writeImpl()
	{
		writeD(_charObjId);
		writeD(_running ? RUN : WALK);
		writeD(0); //c2
	}
}

File: game/src/main/java/l2server/gameserver/network/serverpackets/ChangeMoveType.kt
/*
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
 * details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see <http://www.gnu.org/licenses/>.
 */

package l2server.gameserver.network.serverpackets

import l2server.gameserver.model.actor.Creature

/**
 * sample
 *
 *
 * 0000: 3e 2a 89 00 4c 01 00 00 00						 .|...
 *
 *
 * format   dd
 *
 * @version $Revision: 1.3.2.1.2.4 $ $Date: 2005/03/27 15:29:57 $
 */
class ChangeMoveType(character: Creature) : L2GameServerPacket() {

	private val charObjId: Int
	private val running: Boolean

	init {
		charObjId = character.objectId
		running = character.isRunning
	}

	override fun writeImpl() {
		writeD(charObjId)
		writeD(if (running) RUN else WALK)
		writeD(0) //c2
	}

	companion object {
		val WALK = 0
		val RUN = 1
	}
}
